### PR TITLE
751 dzelge dat op fail unexpectedly

### PR DESCRIPTION
**Pull request by dzelge**
Fixed CRS issue in data_frame_query, data_frame_subset, and data_frame_closest (#751). GeoDataFrames are passed into these ops maintaining their original projection

### CODE REVIEW

File: cate/ops/data_frame.py [hunk 1]
```diff
@@ -250,6 +263,10 @@ def data_frame_find_closest(gdf: gpd.GeoDataFrame,
                     geom_point = geometry.representative_point()
                 except AttributeError as e:
                     raise ValidationError('Invalid geometry column.') from e
+                # Features that span the poles will cause shapely.representative_point() to crash.
+                # The quick and dirty solution was to catch the exception and ignore it
+                except ValueError:
+                    pass
```
Comment (forman) [inline cate/ops/data_frame.py]: Oh, no. But good that you comment that.

File: cate/ops/data_frame.py [hunk 2]
```diff
@@ -230,6 +241,8 @@ def data_frame_find_closest(gdf: gpd.GeoDataFrame,
     location = GeometryLike.convert(location)
     location_point = location.representative_point()
 
+    location_point = _transform_coordinates(location_point, gdf.crs)
```
Comment (forman) [inline cate/ops/data_frame.py]: Check if it should read:

    location_point = _transform_coordinates(location_point, gdf.crs or "EPSG:4326")

File: cate/ops/data_frame.py [hunk 3]
```diff
@@ -114,27 +117,32 @@ def data_frame_query(df: DataFrameLike.TYPE, query_expr: str) -> pd.DataFrame:
     data_frame = DataFrameLike.convert(df)
 
     local_dict = dict(from_wkt=GeometryLike.convert)
-    if hasattr(data_frame, 'geometry') and isinstance(data_frame.geometry, gpd.GeoSeries):
+    if hasattr(data_frame, 'geometry') \
+            and isinstance(data_frame.geometry, gpd.GeoSeries) \
+            and hasattr(data_frame, 'crs'):
+
+        crs = data_frame.crs
```
Comment (forman) [inline cate/ops/data_frame.py]: Check if it should read:

    crs  = crs or "EPSG:4326"

File: cate/ops/data_frame.py [hunk 4]
```diff
@@ -400,8 +417,28 @@ def great_circle_distance(p1: shapely.geometry.Point, p2: shapely.geometry.Point
...
 
 
-def _data_frame_geometry_op(instance_method, geometry: GeometryLike):
+# Transforms a geometry that is used in an operator for e.g. feature selection purposes. It assures
+# that both use an identical 'crs' (projection)
+def _transform_coordinates(geom, crs: dict):
+    if crs and 'init' not in crs:
+        raise ValidationError('No crs in GeoDataFrame' + str(crs))
+
+    if crs and crs['init'] != 'epsg:4326':
```
Comment (forman) [inline cate/ops/data_frame.py]: What about `'EPSG:4326'` (capital letters). Make sure, names are normalised befor comparison.

File: environment.yml [hunk 5]
```diff
@@ -43,6 +43,7 @@ dependencies:
   - shapely >=1.6,<2.0
   - tornado >=5.0,<6.0
   - xarray >=0.10.6
+  - pyproj >= 1.9.5.1
```
Comment (forman) [inline environment.yml]: You are a danger seeker!

Comment (dzelge) [reply environment.yml]: Shall I remove that?

File: cate/ops/data_frame.py (hunk 1 above)
Comment (dzelge) [reply cate/ops/data_frame.py]: Nich schön, ich weiß

File: cate/ops/data_frame.py [hunk 6]
```diff
@@ -400,8 +417,28 @@ def great_circle_distance(p1: shapely.geometry.Point, p2: shapely.geometry.Point
...
 
 
-def _data_frame_geometry_op(instance_method, geometry: GeometryLike):
+# Transforms a geometry that is used in an operator for e.g. feature selection purposes. It assures
+# that both use an identical 'crs' (projection)
+def _transform_coordinates(geom, crs: dict):
+    if crs and 'init' not in crs:
+        raise ValidationError('No crs in GeoDataFrame' + str(crs))
+
+    if crs and crs['init'] != 'EPSG:4326':
```
Comment (forman) [inline cate/ops/data_frame.py]: Gefahr. Ich denke es sollte heißen:

    if crs and 'init' in crs and str(crs['init']).lower() != 'epsg:4326':

File: cate/ops/data_frame.py [hunk 7]
```diff
@@ -400,8 +417,28 @@ def great_circle_distance(p1: shapely.geometry.Point, p2: shapely.geometry.Point
...
+        project = partial(
+            pyproj.transform,
+            pyproj.Proj(init='EPSG:4326'),  # source coordinate system
```
Comment (forman) [inline cate/ops/data_frame.py]: Also dann durchgängig `"epsg:4326"`

**Pull request merged**